src/components/cli.ts

```diff
@@ -229,7 +233,24 @@ export class Cli<ConfigType extends Record<string, unknown>> {
         });
     }
 
-    private startWithConfig(config: ConfigType|null) {
+    private startWithConfig(config: ConfigType|null, configFilename: string) {
+        if (this.opts.onConfigChanged && this.opts.bridgeConfig) {
+            log.info("Will listen for SIGHUP");
+            process.on("SIGHUP",
+                () => {
+                log.info("Got SIGHUP, reloading config file");
+                try {
+                    const newConfig = this.loadConfig(configFilename);
+                    // onConfigChanged is checked above, but for Typescript's sake.
+                    if (this.opts.onConfigChanged) {
+                        this.opts.onConfigChanged(newConfig);
```

Comment:
Do you want to update the `this.bridgeConfig` variable here?

In real life, only some config variables would be hot-reloadable. I would expect that if the new config edits other variables, the bridge should reject the new config file and retain the old one. We can make `onConfigChanged` return a boolean to indicate whether the new config file is accepted, and act based on that.

Reply:
Nope, I don't want to update the config from underneath bridges because it could cause problems. For instance, certain bits of the IRC bridge cannot be reloaded so we wouldn't want to change those variables.

It will be up to the bridge developer to listen for the callback, and explicitly apply changes to the config inside the bridge. It might be a bit more faff, but it's safer.